### PR TITLE
call addRTAnnotation for projects

### PR DESCRIPTION
**Pull request by kinarashah**
https://github.com/rancher/rancher/issues/14704

### CODE REVIEW

**Comment by cjellick** [comment]
@kinarashah could you explain what you are doing here and why in the commit message better?
A good set of guidelines to follow for content: https://chris.beams.io/posts/git-commit/

Can you add a test for this?

**Comment by cjellick** [comment]
@dramich can you review this please

**Comment by dramich** [comment]
@kinarashah This should be moved down to `createProject`. We want the annotation on the project before we create it or it can cause race/update conditions.

**Comment by kinarashah** [comment]
@dramich I'll add `createProjectAnnotation` to get annotation before creating the project. 

But I just wanted to understand if I am missing something, since I'm calling `addRTAnnotation` always before `reconcileCreatorRTB`, how will it cause such conditions?

**Comment by dramich** [comment]
@kinarashah The annotation isn't getting saved on the project yet, it's just being added, so if the project is updated by another thread before you save and get an outdated error the controller will retry and rebuild the annotations. If the defaults changed between runs you will have existing bindings that are not the current default which causes the `required` and `created` to never match. To avoid this we create the annotation with the project so the `required` is a snapshot at project creation time.

**Comment by kinarashah** [comment]
@dramich Thanks a lot for the detailed explanation! I've updated the PR, could you please review?

File: pkg/controllers/management/auth/project_cluster_handler.go [hunk 1]
```diff
@@ -442,6 +447,27 @@ func (m *mgr) reconcileResourceToNamespace(obj runtime.Object, controller string
 	})
 }
 
+func (m *mgr) createProjectAnnotation() (string, error) {
```
Comment (dramich) [inline pkg/controllers/management/auth/project_cluster_handler.go]: A new function isn't needed, use `addRTAnnotation`. The project will just need to be created outside the call to `Create`, passed through `addRTAnnotation` then into create.

Comment (kinarashah) [reply pkg/controllers/management/auth/project_cluster_handler.go]: sorry, I should've realized `addRTAnnotation` would work even before creating project, updated!

**Comment by dramich** [comment]
LGTM

**Comment by cjellick** [comment]
LGTM

**Pull request merged**